### PR TITLE
DROOLS-2946: [DMN Designer] Data-types: Data-type selector broken

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2946

### CODE REVIEW

**Comment by jomarko** [comment]
Sorry, will have a look next week, working overtime already today.

**Comment by manstis** [comment]
@jomarko @karreiro @etirelli Data-type selection is broken without this merged.

Any chance of a review (or unilateral agreement to merge) for 7.1?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/DataTypePickerWidgetTest.java [hunk 1]
```diff
@@ -212,17 +216,23 @@ public void testMakeTypeSelectorForBuiltInType() {
 
         final ArgumentCaptor<String> optionTextCaptor = ArgumentCaptor.forClass(String.class);
         final ArgumentCaptor<String> optionValueCaptor = ArgumentCaptor.forClass(String.class);
+        final ArgumentCaptor<QName> qNameCaptor = ArgumentCaptor.forClass(QName.class);
 
-        doReturn(bit.getName()).when(qNameConverter).toWidgetValue(any(QName.class));
+        //doReturn(bit.getName()).when(qNameConverter).toWidgetValue(any(QName.class));
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/DataTypePickerWidgetTest.java]: Remove commented line.. will do this evening.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/DataTypePickerWidget.java [hunk 2]
```diff
@@ -118,10 +119,26 @@ private void addBuiltInTypes() {
...
         return Optional.of(o);
     }
 
+    QName normaliseBuiltInTypeTypeRef(final QName typeRef) {
+        String namespace = typeRef.getNamespaceURI();
+        String localPart = typeRef.getLocalPart();
+        String prefix = typeRef.getPrefix();
+
+        final Definitions definitions = dmnGraphUtils.getDefinitions();
+        final Optional<String> nsPrefix = definitions.getPrefixForNamespaceURI(namespace);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/DataTypePickerWidget.java]: `dmnGraphUtils.getDefinitions()` can return null, please protect from NPE on this line.

**Comment by manstis** [comment]
Thank-you @jomarko Updated accordingly.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
@manstis The broken test is not related to this PR:
```

**Pull request merged**